### ISSUE TITLE
Can't install CoreStore via Swift Package Manager 3.1

### ISSUE THREAD

**Issue by TimothyMDean** [JohnEstropia/CoreStore#159]
I'm trying to set up a project with Swift Package Manager 3.1, and I'd like to include CoreStore as a dependency. 

I get the following error when I execute the `swift package fetch` command:

```
error: the package has an unsupported layout, unexpected source file(s) found:
/Path/SwiftProject/.build/checkouts/CoreStore.git--3311833321589799163/Sources/CoreDataNativeType.swift,
/Path/SwiftProject/.build/checkouts/CoreStore.git--3311833321589799163/Sources/CoreStore.swift,
/Path/SwiftProject/.build/checkouts/CoreStore.git--3311833321589799163/Sources/CoreStoreError.swift
fix: move the file(s) inside a module
```

**Comment by JohnEstropia** [comment]
Hi, do you have an idea why this error occurs? The files in the error are all included in the module so I'm not sure what 
```
fix: move the file(s) inside a module
```
means here.

**Comment by TimothyMDean** [comment]
I suspect the problem is due to SPM's convention-based source layout assumption. According to the [documentation](https://github.com/apple/swift-package-manager/blob/master/Documentation/Reference.md#source-layouts), each subfolder under the Source folder is assumed to be its own module. If you are creating a single module, you can put all the source directly under `Sources`.

It seems like your source code layout is mixing up the tool by including source files both at the root level under `Sources`, as well as subdirectories that contain more source files. It seems likely that SPM sees the subdirectories and assumes this is a multi-module source layout, but then it doesn't know what to do with the Swift files directly under `Sources`.

That's just my theory though - I haven't dug through the SPM source code to see if my theory is correct.

**Comment by JohnEstropia** [comment]
Ah, that must be it. I've been looking for this documentation.

Hmm... I need to look for a way to tell the package manager to flatten the directories. I don't want to move all files in a single directory.

If anyone finds a way to do this, please let me know.

**Comment by JohnEstropia** [comment]
@TimothyMDean Sorry for the wait. I've pushed a fix in CoreStore 3.1.1. Tell me if it works for you.
